website/views.py

```diff
@@ -38,53 +85,147 @@ def register(request):
             messages.success(request, "Successfully Registered")
             return redirect('home')
         else:
-            return render(request, 'register.html', {'form': form})
-    else:
+            return render(request, self.template_name, {'form': form})
+    def get(self,request):
         form = SignUpForm()
-        return render(request, 'register.html', {'form': form})
-    return render(request, 'register.html', {'form': form})
+        return render(request, self.template_name, {'form': form})
 
+# Function Based Record View
+# def record(request,pk):
+#     if request.user.is_authenticated:
+#         record=Record.objects.get(id=pk)
+#         return render(request,'record.html',{'cust_record':record})
+#     else:
+#         messages.success(request,"First Login or signup please !!!")
+#         return redirect('home')
 
-def record(request,pk):
-    if request.user.is_authenticated:
-        record=Record.objects.get(id=pk)
-        return render(request,'record.html',{'cust_record':record})
-    else:
-        messages.success(request,"First Login or signup please !!!")
-        return redirect('home')
+class record(LoginRequiredMixin, View):
+    template_name='record.html'
+    def get_record(pk):
+        try:
+            return Record.objects.get(id=pk)
+        except:
+            return None
+    def get(self,request,pk):
+        rec= record.get_record(pk)
```

Comment:
`get_record` can be a classmethod e.g
you have Record model, add a @classmethod decorator in that.